### PR TITLE
MRG: Speed up tests

### PR DESCRIPTION
**Pull request by larsoner**
This should speed up our tests a bit. I noticed some were going over 40 minutes and thus flirting with the Travis test limit.

Big speed gains were realized by changing code used by `setup_source_space` to much more efficiently compute the neighboring triangles for each vertex.

Minor speed gains in forward computation came from different vectorization forms.

### CODE REVIEW

File: mne/viz/tests/test_evoked.py [hunk 1]
```diff
@@ -168,21 +182,20 @@ def test_plot_evoked():
...
-    evoked.plot_image(group_by=dict(sel=[0, 7]), axes=dict(sel=plt.axes()))
+    evoked.plot_image(group_by=dict(sel=sel), axes=dict(sel=plt.axes()))
     plt.close('all')
     for group_by, axes in (("something", dict()), (dict(), "something")):
         pytest.raises(ValueError, evoked.plot_image, group_by=group_by,
                       axes=axes)
 
-    # test plot_topo
-    evoked.plot_topo()  # should auto-find layout
-    _line_plot_onselect(0, 200, ['mag', 'grad'], evoked.info, evoked.data,
```
Comment (agramfort) [inline mne/viz/tests/test_evoked.py]: why removing this one?

**Comment by agramfort** [comment]
besides I'll let travis judge the rest

thanks a ton @larsoner

File: mne/viz/tests/test_evoked.py (hunk 1 above)
Comment (larsoner) [reply mne/viz/tests/test_evoked.py]: There is a test_topo file, but I should maybe move it instead of delete

Comment (agramfort) [reply mne/viz/tests/test_evoked.py]: I am just a bit cautious removing test in code but if you tell me it's covered else where fine.

**Pull request merged**